src/select/SelectTrigger.ts

```diff
@@ -1,27 +1,112 @@
-import { useForkRef } from "reakit-utils";
+import React from "react";
 import { BoxHTMLProps } from "reakit/ts/Box/Box";
-import { SelectStateReturn } from "./useSelectState";
+import { SelectStateReturn } from "./SelectState";
 import { createHook, createComponent } from "reakit-system";
-import { SELECT_KEYS, POPOVER_DISCLOSURE_KEYS } from "./__keys";
-import { useDialogDisclosure } from "reakit";
+import { SELECT_KEYS } from "./__keys";
+import { usePopoverDisclosure } from "reakit/Popover";
+import { useShortcut } from "@chakra-ui/hooks";
+import { createOnKeyDown, useLiveRef, useForkRef } from "reakit-utils";
+import { callAllHandlers, getNextItemFromSearch } from "@chakra-ui/utils";
 
 export type SelectTriggerOptions = Pick<
   SelectStateReturn,
-  "toggle" | "visible" | "unstable_referenceRef"
->;
+  | "visible"
+  | "values"
+  | "selected"
+  | "setSelected"
+  | "show"
+  | "hide"
+  | "first"
+  | "last"
+> & { disabled?: boolean };
 
 const useSelectTrigger = createHook<SelectTriggerOptions, BoxHTMLProps>({
   name: "selectTrigger",
-  compose: useDialogDisclosure,
+  compose: [usePopoverDisclosure],
   keys: SELECT_KEYS,
 
-  useProps({ unstable_referenceRef, visible }, { ref: htmlRef, ...htmlProps }) {
+  useProps(
+    options,
+    {
+      ref: htmlRef,
+      onClick: htmlOnClick,
+      onMouseDown: htmlOnMouseDown,
```

Comment:
No more needed !!

Reply:
We do have Eslint setup right? Why isn't mine working? lol